cmd/kubelet-plugin/pci_device.go

```diff
@@ -0,0 +1,118 @@
+package main
+
+import (
+	"log"
+	"os"
+	"path/filepath"
+
+	"github.com/google/uuid"
+)
+
+const (
+	pciBasePath = "/sys/bus/pci/devices"
+	PCIResourcePrefix   = "PCI_RESOURCE"
+)
+
+type PCIDevice struct {
+	uuid           string
+	vendorSelector string
+	resourceName   string
+	pciAddress     string
+	driver         string
+	iommuGroup     string
+	numaNode       int
+	pciID          string
+}
+
+func DiscoverPermittedHostPCIDevices(supportedPCIDeviceMap map[string]string) (map[string][]*PCIDevice) {
```

Comment:
Won't it make sense to return an error as well here?

Reply:
I am not sure , because , in any case we need return a empty `map[string][]*PCIDevice) `, in case there are error while reading the PCIdevice , its logged right now

Reply:
But it is in general best practise to let the caller handle errors